### PR TITLE
Add security token to R1 calls to API

### PR DESCRIPTION
**Pull request by ThomasOSI**
[#139758547]

### CODE REVIEW

**Comment by pemaOSI** [comment]
We will have to pass in security token as header for all API request.

File: spec/repositories/person_repository_spec.rb [hunk 1]
```diff
@@ -91,31 +90,75 @@
...
 
   describe '.search' do
-    it 'raise an error if the response code is not 200' do
-      stub_request(:get, %r{/api/v1/people_search\?search_term=})
-        .and_return(status: 500, headers: { 'Content-Type': 'application/json' })
-
-      expect do
-        described_class.search('')
-      end.to raise_error('Error searching people')
+    json_header = { 'Content-Type': 'application/json' }
```
Comment (NateOSI) [inline spec/repositories/person_repository_spec.rb]: shouldn't json_header contain { 'Authorization': 'SecurityToken' } as well?

File: app/repositories/person_repository.rb [hunk 2]
```diff
@@ -23,26 +23,31 @@ def self.find(id)
 
   def self.update(person)
     raise 'Error updating person: id is required' unless person.id
-    response = make_api_call("#{PEOPLE_PATH}/#{person.id}", :put, person)
+    response = make_api_call("#{PEOPLE_PATH}/#{person.id}", :put, nil, person)
```
Comment (pemaOSI) [inline app/repositories/person_repository.rb]: can we use named parameters?

File: app/controllers/api/v1/people_controller.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@ def update
       end
 
       def search
-        people_results = PersonRepository.search(params[:search_term])
```
Comment (pemaOSI) [inline app/controllers/api/v1/people_controller.rb]: Api.connection.headers['Authorization'] = session['security_token']

**Comment by ThomasOSI** [comment]
This solution was discarded.

**Pull request closed without merging**